data_handler.py

```diff
@@ -0,0 +1,108 @@
+from auxiliaries import datetime_from_epoch
+from sender import ServerSender
+from sensor import Sensor
+from auxiliaries import set_verbosity
+from globalvalues import ANSI_RESET, ANSI_YEL, ANSI_GR, ANSI_RED
+from collections import deque
+import socket
+import time
+
+CPM_DISPLAY_TEXT = (
+    '{{time}}: {yellow} {{counts}} cts{reset}' +
+    ' --- {green}{{cpm:.2f}} +/- {{cpm_err:.2f}} cpm{reset}' +
+    ' ({{start_time}} to {{end_time}})').format(
+    yellow=ANSI_YEL, reset=ANSI_RESET, green=ANSI_GR)
+strf = '%H:%M:%S'
+
+class Data_Handler(object):
+    """
+    Object for sending data to server. 
+
+    Also handles writing to datalog and 
+    storing to memory. 
+    """
+
+    def __init__(self,
+                 manager=None,
+		 verbosity=1,
+		 logfile=None):
+
+        self.v = verbosity
+	if manager and logfile is None:
+	    set_verbosity(self, logfile=manager.logfile)
+	else:
+	    set_verbosity(self, logfile=logfile)
+		
+	self.manager = manager
+    	
+    	self.queue = deque('')
+    	
+    def test_send(self, cpm, cpm_err):
+        """
+	Test Mode
+	"""
+	self.vprint(
+	    1, ANSI_RED + " * Test mode, not sending to server * " +
+	    ANSI_RESET)
+
+    def no_config_send(self, cpm, cpm_err):
+	"""
+	Configuration file not present
+	"""
+	self.vprint(1, "Missing config file, not sending to server")
+
+    def no_publickey_send(self, cpm, cpm_err):
+    	"""
+	Publickey not present
+	"""
+	self.vprint(1, "Missing public key, not sending to server")
+
+    def no_network_send(self, cpm, cpm_err):
+        """
+	Network is not up
+	"""
+	self.vprint(1, "Network down, not sending to server")
+	self.manager.send_to_queue(cpm, cpm_err)
+
+    def regular_send(self, cpm, cpm_err):
+        """
+	Normal send
+	"""
+	try:
+	    self.manager.sender.send_cpm(cpm, cpm_err)
+	    while self.queue:

```

Comment:
We need a way to toggle the "new format" communication off or on for testing with the server. So there should be an input argument controlling this, let's say `--protocol` in `argparse` and `protocol` in `Manager.__init__()`. Similar to before, set the default in argparse to be `None` and put a `DEFAULT_PROTOCOL` in `globalvalues`. The value can be `'old'` or `'new'`.
* `'old'` can still make the memory queue and the backlog file, but these should never be flushed to the server because the communication protocol doesn't support it (no timestamp is given). You can do this nicely here in `regular_send` by doing `while self.queue and self.protocol == 'new'`.
* `'new'` does everything you have here.
* The protocol variable would get passed into `Manager` but then from there passed to `DataHandler` as well as `ServerSender`. Right now `ServerSender` doesn't have a new protocol yet, I'll tell you when that is defined in the next couple days.